statsmodels/multivariate/tests/test_factor.py

```diff
@@ -0,0 +1,129 @@
+
+try:
+    import matplotlib.pyplot as plt
+    missing_matplotlib = False
+except ImportError:
+    missing_matplotlib = True
+
+# Example data
+# https://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/
+#     viewer.htm#statug_introreg_sect012.htm
+X = pd.DataFrame([['Minas Graes', 2.068, 2.070, 1.580, 1, 0],
+                  ['Minas Graes', 2.068, 2.074, 1.602, 2, 1],
+                  ['Minas Graes', 2.090, 2.090, 1.613, 3, 0],
+                  ['Minas Graes', 2.097, 2.093, 1.613, 4, 1],
+                  ['Minas Graes', 2.117, 2.125, 1.663, 5, 0],
+                  ['Minas Graes', 2.140, 2.146, 1.681, 6, 1],
+                  ['Matto Grosso', 2.045, 2.054, 1.580, 7, 0],
+                  ['Matto Grosso', 2.076, 2.088, 1.602, 8, 1],
+                  ['Matto Grosso', 2.090, 2.093, 1.643, 9, 0],
+                  ['Matto Grosso', 2.111, 2.114, 1.643, 10, 1],
+                  ['Santa Cruz', 2.093, 2.098, 1.653, 11, 0],
+                  ['Santa Cruz', 2.100, 2.106, 1.623, 12, 1],
+                  ['Santa Cruz', 2.104, 2.101, 1.653, 13, 0]],
+                 columns=['Loc', 'Basal', 'Occ', 'Max', 'id', 'alt'])
+
+
+def test_specify_nobs():
+    # Test specifying nobs
+    Factor(np.zeros([10, 3]), 2, nobs=10)
+    assert_raises(ValueError, Factor, np.zeros([10, 3]), 2, nobs=9)
+
+
+def test_auto_col_name():
+    # Test auto generated variable names when endog_names is None
+    mod = Factor(None, 2, corr=np.zeros([11, 11]), endog_names=None,
+                 smc=False)
+    assert_array_equal(mod.endog_names,
+                       ['var00', 'var01', 'var02', 'var03', 'var04', 'var05',
+                        'var06', 'var07', 'var08', 'var09', 'var10'])
+
+
+def test_direct_corr_matrix():
+    # Test specifying the correlation matrix directly
+    mod = Factor(None, 2, corr=np.corrcoef(X.iloc[:, 1:-1], rowvar=0),
+                 smc=False)
+    results = mod.fit(tol=1e-10)
+    a = np.array([[0.965392158864, 0.225880658666255],
+                  [0.967587154301, 0.212758741910989],
+                  [0.929891035996, -0.000603217967568],
+                  [0.486822656362, -0.869649573289374]])
+    assert_array_almost_equal(results.loadings, a, decimal=8)
+    # Test set and get endog_names
+    mod.endog_names = X.iloc[:, 1:-1].columns
+    assert_array_equal(mod.endog_names, ['Basal', 'Occ', 'Max', 'id'])
+
+    # Test set endog_names with the wrong number of elements
+    assert_raises(ValueError, setattr, mod, 'endog_names',
+                  X.iloc[:, :1].columns)
+
+
+def test_unknown_fa_method_error():
+    # Test raise error if an unkonwn FA method is specified in fa.method
+    mod = Factor(X.iloc[:, 1:-1], 2, method='ab')
+    assert_raises(ValueError, mod.fit)
+
+
+def test_example_compare_to_R_output():
+    # No rotation without squared multiple correlations prior
+    # produce same results as in R `fa`
+    mod = Factor(X.iloc[:, 1:-1], 2, smc=False)
```

Comment:
Actually, the unit test is here

Reply:
I guess I need better reading glasses.
Thanks

Reply:
No problem, just let me know if there is anything else need to be done